packages/booter-lb3app/src/lb3app.booter.ts

```diff
@@ -0,0 +1,119 @@
+const swagger2openapi = require('swagger2openapi');
+
+const debug = debugFactory('loopback:boot:lb3app');
+
+const DefaultOptions: Lb3AppBooterOptions = {
+  // from "/dist/application.ts" to "/legacy"
+  path: '../legacy/server/server',
+  mode: 'full',
+};
+
+export class Lb3AppBooter implements Booter {
+  options: Lb3AppBooterOptions;
+  appPath: string;
+
+  constructor(
+    @inject(CoreBindings.APPLICATION_INSTANCE)
+    public app: RestApplication,
+    @inject(BootBindings.PROJECT_ROOT)
+    public projectRoot: string,
+    @inject(`${BootBindings.BOOT_OPTIONS}#lb3app`)
+    options: Partial<Lb3AppBooterOptions> = {},
+  ) {
+    this.options = Object.assign({}, DefaultOptions, options);
+
+    if (typeof app.mountExpressRouter !== 'function') {
+      throw new Error(
+        'Lb3AppBooter requires RestApplication with mountExpressRouter API',
+      );
+    }
+  }
+
+  async configure?(): Promise<void> {
+    this.appPath = path.join(this.projectRoot, this.options.path);
+  }
+
+  async load?(): Promise<void> {
+    const legacyApp = await this.loadAndBootTheApp();
+    const spec = await this.buildOpenApiSpec(legacyApp);
+    if (this.options.mode === 'full') {
+      this.mountFullApp(legacyApp, spec);
+    } else {
+      this.mountRoutesOnly(legacyApp, spec);
+    }
+
+    // TODO(bajtos) Listen for the following events to update the OpenAPI spec:
+    // - modelRemoted
+    // - modelDeleted
+    // - remoteMethodAdded
+    // - remoteMethodDisabled
+    // Note: LB4 does not support live spec updates yet.
+  }
+
+  private async loadAndBootTheApp() {
+    debug('Loading LB3 app from', this.appPath);
+    const legacyApp = require(this.appPath);
+
+    if (legacyApp.booting) {
+      // Wait until the legacy LB3 app boots
+      await pEvent(legacyApp, 'booted');
+    }
+
+    return legacyApp as Lb3Application;
+  }
+
+  private async buildOpenApiSpec(legacyApp: Lb3Application) {
+    const swaggerSpec = generateSwaggerSpec(legacyApp);
+    const result = await swagger2openapi.convertObj(swaggerSpec, {
+      // swagger2openapi options
+    });
+    return result.openapi as OpenApiSpec;
+  }
+
+  private mountFullApp(legacyApp: Lb3Application, spec: OpenApiSpec) {
+    const restApiRoot = legacyApp.get('restApiRoot') || '/';
+    debug('Mounting the entire LB3 app at %s', restApiRoot);
+    const specInRoot = rebaseOpenApiSpec(spec, restApiRoot);
+    this.app.mountExpressRouter('/', legacyApp, specInRoot);
+  }
+
+  private mountRoutesOnly(legacyApp: Lb3Application, spec: OpenApiSpec) {
+    // TODO(bajtos) make this configurable
+    const restApiRoot = '/api';
+    debug('Mounting LB3 REST router at %s', restApiRoot);
+    this.app.mountExpressRouter(
+      restApiRoot,
+      // TODO(bajtos) reload the handler when a model/method was added/removed
+      legacyApp.handler('rest'),
+      spec,
+    );
+  }
+}
+
+export interface Lb3AppBooterOptions {
+  path: string;
+  mode: 'full' | 'router';
```

Comment:
"full" mode means that the entire LB3 application is mounted on LB4, including Express middleware configured in `server/middleware.json`, e.g. `helmet` and `loopback#token`.

"router" mode means that only REST routes (`app.handler('rest')`) are mounted on LB4, no LB3 request-preprocessing middleware is executed.

Reply:
Maybe `'full' | 'restOnly'`.

Reply:
How about `'fullApp' | 'restRouter'`?